subcommands/config/device_group.go

```diff
@@ -0,0 +1,128 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cheynewallace/tabby"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func init() {
+	groupCmd := &cobra.Command{
+		Use:   "device-group",
+		Short: "Manage factory device groups",
+	}
+	cmd.AddCommand(groupCmd)
+
+	groupCmd.AddCommand(&cobra.Command{
+		Use:   "list",
+		Short: "Show available device groups",
+		Run:   doListDeviceGroup,
+	})
+	groupCmd.AddCommand(&cobra.Command{
+		Use:   "create <name> [<description>]",
+		Short: "Create a new device groups",
+		Run:   doCreateDeviceGroup,
+		Args:  cobra.RangeArgs(1, 2),
+	})
+	groupCmd.AddCommand(&cobra.Command{
+		Use:   "delete <name>",
+		Short: "Delete an existing device group",
+		Run:   doDeleteDeviceGroup,
+		Args:  cobra.ExactArgs(1),
+	})
+
+	updateCmd := &cobra.Command{
+		Use:   "update <name>",
+		Short: "Rename an existing device group",
+		Run:   doUpdateDeviceGroup,
+		Args:  cobra.ExactArgs(1),
+	}
+	groupCmd.AddCommand(updateCmd)
+	updateCmd.Flags().StringP("name", "n", "", "Change a device name")
+	updateCmd.Flags().StringP("description", "d", "", "Change a device description")
+}
+
+func doListDeviceGroup(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	logrus.Debugf("Showing a list of device groups for %s", factory)
+
+	lst, err := api.FactoryListDeviceGroup(factory)
+	if err != nil {
```

Comment:
Looks good. There's a new helper I did to deal with these commands exiting on err!=nil:

 https://github.com/foundriesio/fioctl/blob/master/subcommands/common.go#L159

that fixes a bunch of inconsistent ways this has been done. Including here where there's a space in "ERROR: "  - golang will do a space as well so you wind up with 2 spaces instead of one.

Reply:
Cool, used it